oem/ibm/libpldmresponder/utils.cpp

```diff
@@ -204,25 +204,95 @@ bool checkIfIBMFru(const std::string& objPath)
     return false;
 }
 
-std::vector<std::string> findPortObjects(const std::string& adapterObjPath)
+bool checkIfIBMCableCard(const std::string& objPath)
 {
-    using ItemConnector =
-        sdbusplus::client::xyz::openbmc_project::inventory::item::Connector<>;
+    constexpr auto pcieAdapterModelInterface =
+        "xyz.openbmc_project.Inventory.Decorator.Asset";
+    constexpr auto modelProperty = "Model";
 
-    std::vector<std::string> portObjects;
     try
     {
-        portObjects = pldm::utils::DBusHandler().getSubTreePaths(
-            adapterObjPath, 0,
-            std::vector<std::string>({ItemConnector::interface}));
+        auto propVal = pldm::utils::DBusHandler().getDbusPropertyVariant(
+            objPath.c_str(), modelProperty, pcieAdapterModelInterface);
+        const auto& model = std::get<std::string>(propVal);
+        if (!model.empty())
+        {
+            return true;
+        }
+    }
+    catch (const sdbusplus::exception::SdBusError&)
+    {
+        return false;
+    }
+    return false;
+}
+
+void findPortObjects(const std::string& cardObjPath,
+                     std::vector<std::string>& portObjects)
+{
+    static constexpr auto MAPPER_BUSNAME = "xyz.openbmc_project.ObjectMapper";
+    static constexpr auto MAPPER_PATH = "/xyz/openbmc_project/object_mapper";
```

Comment:
you could use PDI.

Reply:
I made some of the PDI changes and the testing failed. I will change this after a discussion

Reply:
done